data/data/gcp/network/network.tf

```diff
@@ -1,58 +1,71 @@
   name          = "${var.cluster_id}-worker-subnet"
-  network       = google_compute_network.cluster_network.self_link
+  network       = google_compute_network.cluster_network[0].self_link
   ip_cidr_range = var.worker_subnet_cidr
+  count         = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_subnetwork" "master_subnet" {
   name          = "${var.cluster_id}-master-subnet"
-  network       = google_compute_network.cluster_network.self_link
+  network       = google_compute_network.cluster_network[0].self_link
   ip_cidr_range = var.master_subnet_cidr
+  count         = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_router" "router" {
   name    = "${var.cluster_id}-router"
-  network = google_compute_network.cluster_network.self_link
+  network = google_compute_network.cluster_network[0].self_link
+  count   = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_address" "master_nat_ip" {
-  name = "${var.cluster_id}-master-nat-ip"
+  name  = "${var.cluster_id}-master-nat-ip"
+  count = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_router_nat" "master_nat" {
   name                               = "${var.cluster_id}-nat-master"
-  router                             = google_compute_router.router.name
+  router                             = google_compute_router.router[0].name
   nat_ip_allocate_option             = "MANUAL_ONLY"
-  nat_ips                            = [google_compute_address.master_nat_ip.self_link]
+  nat_ips                            = [google_compute_address.master_nat_ip[0].self_link]
   min_ports_per_vm                   = 7168
   source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
+  count                              = var.preexisting_cluster_network == "" ? 1 : 0
 
   subnetwork {
-    name                    = google_compute_subnetwork.master_subnet.self_link
+    name                    = google_compute_subnetwork.master_subnet[0].self_link
     source_ip_ranges_to_nat = ["ALL_IP_RANGES"]
   }
 }
 
 resource "google_compute_address" "worker_nat_ip" {
-  name = "${var.cluster_id}-worker-nat-ip"
+  name  = "${var.cluster_id}-worker-nat-ip"
+  count = var.preexisting_cluster_network == "" ? 1 : 0
 }
 
 resource "google_compute_router_nat" "worker_nat" {
   name                               = "${var.cluster_id}-nat-worker"
-  router                             = google_compute_router.router.name
+  router                             = google_compute_router.router[0].name
   nat_ip_allocate_option             = "MANUAL_ONLY"
-  nat_ips                            = [google_compute_address.worker_nat_ip.self_link]
+  nat_ips                            = [google_compute_address.worker_nat_ip[0].self_link]
   min_ports_per_vm                   = 128
   source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
+  count                              = var.preexisting_cluster_network == "" ? 1 : 0
```

Comment:
all count values should be the first field and be separated from the rest by a new line (separate and clear section) for readability.